pkg/openstack/dataplane.go

```diff
@@ -65,7 +66,65 @@ func DataplaneNodesetsDeployed(version *corev1beta1.OpenStackVersion, dataplaneN
 		if nodeset.Status.DeployedVersion != version.Spec.TargetVersion {
 			return false
 		}
+	}
+	return true
+}
 
+// DataplaneNodesetsUpdated returns true if all nodesets are deployed with the latest version
+func DataplaneNodesetsUpdated(ctx context.Context, helper *helper.Helper, version *corev1beta1.OpenStackVersion, dataplaneNodesets *dataplanev1.OpenStackDataPlaneNodeSetList) bool {
```

Comment:
this will get called repeatedly during a minor update until it reconciles.  while this may work in some cases it also depends on DataplaneNodeSet to trigger a reconcile loop on OpenStackVersion (this is the only resource we are watching) and it is not as efficient as perhaps just having the OpenStackDataplaneNodeset controller do it (since it already watches Deployment anyway).

Reply:
Ok, I have reworked it to set a condition on the NodeSet instead, and just updated the func here to check that condition.